xmlua/libxml2/xml-io.lua

```diff
@@ -0,0 +1,50 @@
+local ffi = require("ffi")
+
+ffi.cdef[[
+
+/**
+ * xmlOutputWriteCallback:
+ * @context:  an Output context
+ * @buffer:  the buffer of data to write
+ * @len:  the length of the buffer in bytes
+ *
+ * Callback used in the I/O Output API to write to the resource
+ *
+ * Returns the number of bytes written or -1 in case of error
+ */
+ typedef int (*xmlOutputWriteCallback) (void * context, const char * buffer,
+ int len);
+/**
+* xmlOutputCloseCallback:
+* @context:  an Output context
+*
+* Callback used in the I/O Output API to close the resource
+*
+* Returns 0 or -1 in case of error
+*/
+typedef int (*xmlOutputCloseCallback) (void * context);
+
+struct _xmlOutputBuffer {
+  void*                   context;
+  xmlOutputWriteCallback  writecallback;
+  xmlOutputCloseCallback  closecallback;
+
+  xmlCharEncodingHandlerPtr encoder; /* I18N conversions to UTF-8 */
+
+  xmlBufPtr buffer;    /* Local buffer encoded in UTF-8 or ISOLatin */
+  xmlBufPtr conv;      /* if encoder != NULL buffer for output */
```

Comment:
this uses xmlBufPtr, which seems to be a newer version of xmlBufferPtr, since version 2.9.0.

Reply:
We can assume libxml2 2.9.0 or later because AlmaLinux 8 ships 2.9.7 and Ubuntu 20.04 ships 2.9.10.